Converter/src/indexer.cpp

```diff
@@ -3,1297 +3,1613 @@
-		}
+            indexer->octreeDepth = std::max(indexer->octreeDepth, octreeDepth);
+        }
 
-		if (subject->numPoints == numPoints) {
-			// the subsplit has the same number of points than the input -> ERROR
+        int64_t sanityCheck = 0;
+        for (int64_t nodeIndex = 0; nodeIndex < needRefinement.size(); nodeIndex++) {
+            auto subject = needRefinement[nodeIndex];
+            auto buffer = subject->points;
 
-			unordered_map<string, int> counters;
+            if (sanityCheck > needRefinement.size() * 2) {
+                logger::ERROR("failed to partition point cloud in indexer::buildHierarchy().");
+            }
 
-			auto bpp = attributes.bytes;
+            if (subject->numPoints == numPoints) {
+                // the subsplit has the same number of points than the input -> ERROR
 
-			for (int64_t i = 0; i < numPoints; i++) {
+                unordered_map<string, int> counters;
 
-				int64_t sourceOffset = i * bpp;
+                auto bpp = attributes.bytes;
 
-				int32_t X, Y, Z;
-				memcpy(&X, buffer->data_u8 + sourceOffset + 0, 4);
-				memcpy(&Y, buffer->data_u8 + sourceOffset + 4, 4);
-				memcpy(&Z, buffer->data_u8 + sourceOffset + 8, 4);
+                for (int64_t i = 0; i < numPoints; i++) {
 
-				stringstream ss;
-				ss << X << ", " << Y << ", " << Z;
+                    int64_t sourceOffset = i * bpp;
 
-				string key = ss.str();
-				counters[key]++;
-			}
+                    int32_t X, Y, Z;
+                    memcpy(&X, buffer->data_u8 + sourceOffset + 0, 4);
+                    memcpy(&Y, buffer->data_u8 + sourceOffset + 4, 4);
+                    memcpy(&Z, buffer->data_u8 + sourceOffset + 8, 4);
 
-			int64_t numPointsInBox = subject->numPoints;
-			int64_t numUniquePoints = counters.size();
-			int64_t numDuplicates = numPointsInBox - numUniquePoints;
+                    stringstream ss;
+                    ss << X << ", " << Y << ", " << Z;
 
-			if (numDuplicates < maxPointsPerChunk / 2) {
-				// few uniques, just unfavouribly distributed points
-				// print warning but continue
+                    string key = ss.str();
+                    counters[key]++;
+                }
 
-				stringstream ss;	
-				ss << "Encountered unfavourable point distribution. Conversion continues anyway because not many duplicates were encountered. ";
-				ss << "However, issues may arise. If you find an error, please report it at github. \n";
-				ss << "#points in box: " << numPointsInBox << ", #unique points in box: " << numUniquePoints << ", ";
-				ss << "min: " << subject->min.toString() << ", max: " << subject->max.toString();
+                int64_t numPointsInBox = subject->numPoints;
+                int64_t numUniquePoints = counters.size();
+                int64_t numDuplicates = numPointsInBox - numUniquePoints;
 
-				logger::WARN(ss.str());
-			} else {
+                if (numDuplicates < maxPointsPerChunk / 2) {

```

Comment:
why is. maxPointsPerChunk/2 a suitble measure of "few" duplicates?

Reply:
No idea. This is part of the original code. Actulaly I made no changes here. May be it is showing a diff due to some code formatting change